src/svd/path.py

```diff
@@ -0,0 +1,298 @@
+#
+# Copyright (c) 2023 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+"""
+Classes for refererencing SVD elements based on name.
+"""
+
+from __future__ import annotations
+
+import re
+from abc import ABC, abstractmethod
+from functools import singledispatch
+from itertools import chain
+from typing import (
+    Any,
+    Generic,
+    Iterable,
+    List,
+    Optional,
+    Sequence,
+    Tuple,
+    TypeVar,
+    Union,
+    overload,
+)
+from typing_extensions import Self
+
+
+PartT = TypeVar("PartT")
```

Comment:
nit: I might be a bit slow today but I can't easily tell what this type stands for when looking at the code, I just assume it's something like "Part Type". Maybe a small comment to say what this type means for extra readability?